dialogflow/spec/context_management_spec.rb

```diff
@@ -0,0 +1,59 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "rspec"
+require "google/cloud/dialogflow"
+
+require_relative "../context_management"
+
+describe "Context Management" do
+
+  before do
+    @project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+    @session_id = "fake_session_for_testing"
+    @context_id = "fake_context_for_testing"
+  end
+
+  example "create context" do
+    expect {
+      list_contexts project_id: @project_id, session_id: @session_id
+    }.not_to output(
+      /#{@context_id}/
+    ).to_stdout
+    expect {
+      create_context project_id: @project_id,
+                     session_id: @session_id,
+                     context_id: @context_id
+    }.to output(
+      /#{@session_id}.*#{@context_id}/m
+    ).to_stdout
+    expect {
+      list_contexts project_id: @project_id, session_id: @session_id
+    }.to output(
+      /#{@context_id}/
+    ).to_stdout
+  end
+
+  example "delete context" do
+    delete_context project_id: @project_id,
```

Comment:
**tl;dr** *These tests run in random order so `example "create context"` may not have run before this, so there is no context to delete. Create a context right above this and you're good to go!* 👍 

----
`<rant>`

There's no Context to delete here.

Seeing the `delete`, I naturally looked for:
 1. Create context in `example()`
 1. Create context in `before()`

But found none 😥 

Ruby tests are run in random order _(this has become the default for the 2 popular testing frameworks, iirc)_ so you can't guarantee this will have anything to delete. The above `example "create context"` may not have been run yet.

Because of this, it's natural for Rubyists to create prerequisite objects either:
 - (a) in the `example` _(aka `it`)_ block . 
 - (b) in `before` so it runs before every test for scenarios where every test is going to operate on a new instance of an object. There is also an `after`.
 - (c) in `before :all do ... end` which runs *once* before all of the `example`s are run, eg. to create some objects multiple tests require and the tests promise not to mutate these objects. There is also an `after :all`.

`</rant>`

Reply:
done.  

thanks for pointing out that the tests run in random order (why is that?)